source/agora/consensus/EnrollmentManager.d

```diff
@@ -711,6 +668,27 @@ public class EnrollmentManager
         return this.cycle_preimages[height];
     }
 
+    /***************************************************************************
+
+        Create signature noise for enrollment data
+
+        Params:
+            height = the height used for a salt value
+
+        Returns:
+            the signature noise created using the private key
+
+    ***************************************************************************/
+
+    private Pair createSignatureNoise (ulong height) nothrow
```

Comment:
```suggestion
    private Pair createSignatureNoise (ulong height) nothrow @safe
```

Might be able to mark it `@nogc` too, although not sure

Reply:
Hu ?

Reply:
I'm not ready for review yet. :)

Reply:
Marking the function `@safe` emits error like `Error: @safe function agora.consensus.EnrollmentManager.EnrollmentManager.createSignatureNoise cannot call @system constructor agora.common.crypto.ECC.Scalar.this`. Do you mean I should modify the code in order to mark the function `@safe`?